calico_kubernetes/tests/policy_test.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+from netaddr import IPNetwork
+from calico_kubernetes import policy
+from calico_kubernetes.policy import PolicyParseError
+
+
+class PolicyParserTest(unittest.TestCase):
+
+    def setUp(self):
+        self.namespace = "testNamespace"
+        self.parser = policy.PolicyParser(self.namespace)
+
+    def test_parse_basic_rule(self):
+        rule = "allow from cidr 172.24.114.0/24 to cidr 1.2.3.4"
+
+        # Parse the rule
+        parsed = self.parser.parse_line(rule)
+
+        # Assert on the result
+        self.assertEqual(parsed['action'], "allow")
+        self.assertEqual(parsed['src_net'], IPNetwork("172.24.114.0/24"))
+        self.assertEqual(parsed['dst_net'], IPNetwork("1.2.3.4/32"))
+
+        rule = "allow from label A=B"
+
+        # Parse the rule
+        parsed = self.parser.parse_line(rule)
+
+        # Assert on the result
+        expected_tag = "%s_A_B" % self.namespace
+        self.assertEqual(parsed['action'], "allow")
+        self.assertEqual(parsed['src_tag'], expected_tag)
+
+    def test_parse_allow(self):
+        rule = "allow"
+
+        # Parse the rule
+        parsed = self.parser.parse_line(rule)
+
+        # Assert on the result
+        self.assertEqual(parsed['action'], "allow")
```

Comment:
Worth throwing in a negative test or two, to check that we don't have an icmp_port, dst_net, etc? I don't think you have to do that on every TC, but would be good to sanity-check at least once.